acceptance/tests/lookup/merge_strategies.rb

```diff
@@ -0,0 +1,194 @@
+test_name 'C99903: merge strategies' do
+  require 'puppet/acceptance/environment_utils.rb'
+  extend Puppet::Acceptance::EnvironmentUtils
+
+  app_type        = File.basename(__FILE__, '.*')
+  tmp_environment = mk_tmp_environment_with_teardown(master, app_type)
+  fq_tmp_environmentpath  = "#{environmentpath}/#{tmp_environment}"
+  tmp_environment2 = mk_tmp_environment_with_teardown(master, app_type)
+  fq_tmp_environmentpath2  = "#{environmentpath}/#{tmp_environment2}"
+
+  master_confdir = master.puppet('master')['confdir']
+  hiera_conf_backup = master.tmpfile(app_type)
+
+  teardown do
+    step "restore default global hiera.yaml" do
+      on(master, "mv #{hiera_conf_backup} #{master_confdir}/hiera.yaml")
```

Comment:
yeah, but you can't do that here without reloading/restarting the server.  i'm trying to avoid that in our tests so we can later remove `#with_puppet_running_on`.

i agree it's a tradeoff and we should probably make a decision on if you can touch the default install/config.  we might even make it difficult to do, in the tooling or on the SUT in the pre-suite steps.

we should do some profiling with the proposed beaker profiler